catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/RegistryPublicationServiceImpl.java

```diff
@@ -142,7 +142,11 @@ public void update(Metacard metacard) throws FederationAdminException {
                 .collect(Collectors.toCollection(HashSet::new));
 
         if (CollectionUtils.isNotEmpty(locations)) {
-            federationAdminService.updateRegistryEntry(metacard, locations);
+            try {
+                federationAdminService.updateRegistryEntry(metacard, locations);
+            } catch (FederationAdminException e) {
+                federationAdminService.addRegistryEntry(metacard, locations);
```

Comment:
It is not the only reason that an exception could be thrown but is the most likely reason. addRegistryEntry shouldn't be a problem in any case though. This was just the simplest way to do this for an exceptional case. If we wanted to avoid doing this we would have to perform a remote query prior to every update to check for an 'exceptional' case. I can change it if you feel that we should be checking first.